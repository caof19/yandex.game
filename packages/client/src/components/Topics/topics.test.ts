import { render } from "@testing-library/react";
import { Topics, TTopic } from ".";
import { BrowserRouter } from "react-router-dom";
const topicTestData: TTopic[] = [
    {
        id: 0,
        title: "mechanical organizer monitored classic ea",
        text: "heard functions russia vb assigned",
        createdAt: "2017-04-01T15:20:04.685Z",
        author: "username651651",
        comments: [],
        reactions: [],
    },
];
it.skip("Список тем форума рендерится без ошибок", () => {
    const { container } = render(
        Topics({
            topics: topicTestData,
            setUpdate: () => {
                console.log();
            },
        }),
        {
            wrapper: BrowserRouter,
        },
    );
    expect(container).toMatchSnapshot();
});

it.skip("Тестовые данные отрисовываются в списке тем форума", () => {
    const { getByText } = render(
        Topics({
            topics: topicTestData,
            setUpdate: () => {
                console.log();
            },
        }),
        {
            wrapper: BrowserRouter,
        },
    );
    expect(getByText(topicTestData[0].title)).toBeTruthy();
});
