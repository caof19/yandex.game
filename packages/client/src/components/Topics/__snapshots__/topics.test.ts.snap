// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Список тем форума рендерится без ошибок 1`] = `
<div>
  <div
    class="ant-list ant-list-split css-dev-only-do-not-override-49qm"
  >
    <div
      class="ant-spin-nested-loading css-dev-only-do-not-override-49qm"
    >
      <div
        class="ant-spin-container"
      >
        <li
          class="ant-list-item"
        >
          <div
            class="ant-card ant-card-bordered css-dev-only-do-not-override-49qm"
          >
            <div
              class="ant-card-head"
            >
              <div
                class="ant-card-head-wrapper"
              >
                <div
                  class="ant-card-head-title"
                >
                  mechanical organizer monitored classic ea
                </div>
                <div
                  class="ant-card-extra"
                >
                  <a
                    href="/forum/0"
                  >
                    Перейти
                  </a>
                </div>
              </div>
            </div>
            <div
              class="ant-card-body"
            >
              <div
                class="ant-flex css-dev-only-do-not-override-49qm ant-flex-justify-space-between"
              >
                <span
                  class="ant-typography css-dev-only-do-not-override-49qm"
                >
                  heard functions russia vb assigned
                </span>
                <span
                  class="ant-typography css-dev-only-do-not-override-49qm"
                >
                  Sat Apr 01 2017
                </span>
              </div>
            </div>
          </div>
        </li>
      </div>
    </div>
  </div>
</div>
`;
